skytemple_files/common/ppmdu_config/pmdsky_debug/loader.py

```diff
@@ -0,0 +1,308 @@
+        if 'length' in definition:
+            if region in definition['length']:
+                binary.length = int(definition['length'][region])
+        if 'functions' in definition:
+            assert binary.loadaddress is not None
+            for symbol_def in definition['functions']:
+                sym = _read_symbol(binary.loadaddress, symbol_def, Pmd2BinarySymbolType.FUNCTION, region)
+                if sym is not None:
+                    binary.symbols.append(sym)
+        if 'data' in definition:
+            assert binary.loadaddress is not None
+            for symbol_def in definition['data']:
+                sym = _read_symbol(binary.loadaddress, symbol_def, Pmd2BinarySymbolType.DATA, region)
+                if sym is not None:
+                    binary.symbols.append(sym)
+        if 'description' in definition:
+            if region in definition['description']:
+                if binary.description == "":
+                    binary.description = definition['description']
+
+
+def _build(binaries: Dict[str, _IncompleteBinary]) -> List[Pmd2Binary]:
+    out = []
+    for bin_name, incl_bin in binaries.items():
+        if incl_bin.loadaddress is None:
+            raise ValueError(f"Did not find a load address for {bin_name}.")
+        if incl_bin.length is None:
+            raise ValueError(f"Did not find a length for {bin_name}.")
+        translated_bin_name = f"{bin_name}.bin"
+        overlay_match = OVERLAY_REGEX.match(bin_name)
+        if overlay_match:
+            translated_bin_name = f"overlay/overlay_{int(overlay_match.group(1)):04}.bin"
+        alias_symbols: List[Pmd2BinarySymbol] = []
+        for symbol in incl_bin.symbols:
+            # Provide aliases for some symbols for backwards compatibility.
+            if symbol.name in ALIAS_MAP:
+                for entry in ALIAS_MAP[symbol.name]:
+                    alias_symbols.append(DeprecatedPmd2BinarySymbol(
+                        name=entry, begin=symbol.begin, end=symbol.end, description=symbol.description, typ=symbol.type,
+                        new_name=symbol.name
+                    ))
+        binary = Pmd2Binary(
+            filepath=translated_bin_name,
+            loadaddress=incl_bin.loadaddress,
+            length=incl_bin.length,
+            symbols=incl_bin.symbols + alias_symbols,
+            description=incl_bin.description
+        )
+        for symbol in binary.symbols.values():
+            symbol.parent = binary
+        out.append(binary)
+
+    return out
+
+
+def load_binaries(edition: str) -> List[Pmd2Binary]:
+    version, region = edition.split("_")
+    if version != GAME_VERSION_EOS:
+        raise ValueError("This game version is not supported.")
+
+    binaries: Dict[str, _IncompleteBinary] = {}
+
+    files = glob(os.path.join(SYMBOLS_DIR, '*.yml'))
+    files.sort(key=lambda key: 1 if key.startswith('arm') or key.startswith('overlay') else -1)
```

Comment:
Am I reading this right? The `sort` method sorts ascending, so assigning 1 to `arm*` and `overlay*` and -1 to everything else is going to put `arm*` and `overlay*` last, not first. I think we want them first so the descriptions make it in properly?

Reply:
Oh, you're right. To be honest I just implemented this blindly from memory. I didn't check that `literals.yml` and `ram.yml` also contain the length and addresses (I just asumed they didn't) - That's the main reason I added this, to make sure those are set first. But guess it still works since those two files still contain everything needed.